web/src/pm_list.ts

```diff
@@ -203,12 +206,26 @@ function zoom_in(): void {
 
 function zoom_out(): void {
     zoomed = false;
-    update_private_messages();
+    if (!clear_search()) {
+        update_private_messages();
+    }
     $(".direct-messages-container").removeClass("zoom-in").addClass("zoom-out");
     $("#streams_list").show();
     $(".left-sidebar .right-sidebar-items").show();
 }
 
+export function clear_search(): boolean {
+    const $filter = $(".pm-list-filter").expectOne();
+    if ($filter.val() !== "") {
+        $filter.val("");
+        update_private_messages();
+        return true;
+    }
+    return false;
+}
+
+const throttled_update_private_message = _.throttle(update_private_messages, 50);
```

Comment:
I'm using `50` ms here since that's what was being used in the channel list filter.

Reply:
I'm not convinced it's actually necessary here; the data set sizes are smaller than in the stream/topics lists, but I expect it to be harmless and maybe save us if the rerenders become expensive due to some future feature.